services/src/main/java/org/keycloak/theme/ResourceLoader.java

```diff
@@ -0,0 +1,51 @@
+package org.keycloak.theme;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Path;
+
+public class ResourceLoader {
+
+    public static InputStream getResourceAsStream(String root, String resource) throws IOException {
+        if (root == null || resource == null) {
+            return null;
+        }
+        Path rootPath = Path.of(root);
+        Path resourcePath = rootPath.resolve(resource).normalize();
+        if (isChild(rootPath, resourcePath)) {
+            URL url = classLoader().getResource(resourcePath.toString());
+            return url != null ? url.openStream() : null;
+        } else {
+            return null;
+        }
+    }
+
+    public static InputStream getFileAsStream(File root, String resource) throws IOException {
+        File file = getFile(root, resource);
+        return file != null && file.isFile() ? file.toURI().toURL().openStream() : null;
+    }
+
+    public static File getFile(File root, String resource) throws IOException {
+        if (root == null || resource == null) {
+            return null;
+        }
+        Path rootPath = root.toPath().normalize();
+        Path resourcePath = rootPath.resolve(resource).normalize();
+        if (isChild(rootPath, resourcePath)) {
+            return resourcePath.toFile();
+        } else {
+            return null;
+        }
+    }
+
+    private static boolean isChild(Path rootPath, Path resourcePath) {
+        return resourcePath.startsWith(rootPath.toString());
```

Comment:
Seems that `startsWith()` can accept a `Path` so perhaps a `toString()` is not needed? 
```suggestion
        return resourcePath.startsWith(rootPath);
```

This probably doesn't need a whole seperate method considering it's a one-liner and is only used twice.